tests/test_all.py

```diff
@@ -93,25 +90,34 @@ def _assert_enabled(self, spider,
         crawler = self._mock_crawler(spider, settings)
         mw = self.mwcls.from_crawler(crawler)
         mw.open_spider(spider)
+        httpproxy = HttpProxyMiddleware.from_crawler(crawler)
```

Comment:
Most test changes are to apply this, i.e. run the Scrapy middleware not just the proxy middleware. They are not specifically needed for this change in the latest iteration, but I think it is best to keep the changes. They e.g. pointed to some bad proxy URLs (missing : after the API key).